File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int *chars = new int[26]();
        
        int left = 0, right = 0;
        int maxLength = 0;
        int maxxFreq = 0;

        while (right < s.size()) {
            maxxFreq = max(maxxFreq, ++chars[s[right] - 'A']);
            if ((right-left+1) - maxxFreq > k) {
                chars[s[left] - 'A']--;
                left++;
            }
            maxLength = max(maxLength, right-left+1);
            right++;
        }
        return maxLength;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        int[] freq = new int[26];
        int left = 0, right = 0;
        int maxLength = int.MinValue;
        int maxFreq = -1;
        while (right < s.Length) {
            maxFreq = Math.Max(maxFreq, ++freq[s[right] - 'A']);
            if ((right - left + 1) - maxFreq > k) --freq[s[left++] - 'A'];
            maxLength = Math.Max(maxLength, (right - left + 1));
            right++;
        }
        return maxLength;
    }
}